### PR TITLE
Add assertRedirectNotContains

### PR DESCRIPTION
**Pull request by gemal**
Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.

The best way to propose a feature is to open an issue first and discuss your ideas there before implementing them.

Always follow the [contribution guidelines](https://github.com/cakephp/cakephp/blob/master/.github/CONTRIBUTING.md) when submitting a pull request. In particular, make sure existing tests still pass, and add tests for all new behavior. When fixing a bug, you may want to add a test to verify the fix.

### CODE REVIEW

**Comment by gemal** [comment]
@ADmad can you review and possibly merge?

**Comment by jeremyharris** [comment]
This is a new feature so shouldn't it be added to `3.next`? Constraints have changed there; this way of asserting isn't used anymore and this new feature would need to be re-done for `3.next`.

File: src/TestSuite/IntegrationTestCase.php [hunk 1]
```diff
@@ -729,6 +729,25 @@ public function assertRedirectContains($url, $message = '')
         $this->assertContains($url, $result['Location'], $message);
     }
 
+    /**
+     * Asserts that the Location header does not contain a substring
+     *
+     * @param string $url The URL you expected the client to go to.
+     * @param string $message The failure message that will be appended to the generated message.
+     * @return void
```
Comment (markstory) [inline src/TestSuite/IntegrationTestCase.php]: Should we have an `@since` tag here?

**Comment by markstory** [comment]
Moved to 3.next as that is where new features go.

**Comment by jeremyharris** [comment]
@gemal let me know if you have trouble migrating over to the new constraints. You should be able to copy the current header contains assertion and use a new `HeaderNotContains` constraint, which you can copy from `HeaderContains` and reverse the logic.

**Comment by saeideng** [comment]
you should rebase on 3.next and push again 
travis now used 3.6 core code as source , this will fix in next push

**Comment by markstory** [comment]
Moved to #12464 as I didn't want to push updates to `gemal:master`.

**Pull request closed without merging**